canopeum_backend/canopeum_backend/serializers.py

```diff
@@ -255,10 +255,14 @@ class Meta:
 class SitetreespeciesSerializer(serializers.ModelSerializer[Sitetreespecies]):
     en = serializers.SerializerMethodField()
     fr = serializers.SerializerMethodField()
+    type_id = serializers.SerializerMethodField()
 
     class Meta:
         model = Sitetreespecies
-        fields = ("id", "quantity", "en", "fr")
+        fields = ("id", "type_id", "quantity", "en", "fr")
+
+    def get_type_id(self, obj) -> int:
+        return TreeTypeSerializer(obj.tree_type).data.get("id", None)  # type: ignore[no-any-return]
```

Comment:
Yeah not much we can do here, `data` is a `dict[Unknown, Unknown]` (`dict[Any, Any] in mypy`). Even if you explicitly type the `obj`